atef/widgets/config/utils.py

```diff
@@ -1402,6 +1405,68 @@ def set_widget_font_size(widget: QWidget, size: int):
     widget.setFont(font)
 
 
+_float_re = re.compile(r'(([+-]?\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?)')
+
+
+def valid_float_string(string):
+    match = _float_re.search(string)
+    return match.groups()[0] == string if match else False
+
+
+class FloatValidator(QValidator):
+
+    def validate(self, string: str, position: int) -> Tuple[QValidator.State, str, int]:
+        if valid_float_string(string):
+            return (QValidator.Acceptable, string, position)
+        if string == "" or string[position-1] in 'e.-+':
```

Comment:
Validate is a method of entry widgets, `string` is the string in the widget, and `position` is the cursor position in that string.  This clause is what lets you type out "2e" or "2." and not have the validator block your inputs.  `position` will be in `range(len(string))`.  

A closer look at this reveals weird ways to break this logic.  I'll try to touch it up

Reply:
This logic branch isn't necessarily perfect, but in cases where you type an invalid float (say: "2.3ee", where the validator still reports "intermediate"), the spinbox will revert the change if you hit enter/shift focus.  I actually think this is ok as is, since the `fixup` method will repair broken entries.